src/compiler/utilities.ts

```diff
@@ -467,16 +467,16 @@ namespace ts {
         return info.declaration ? declarationNameToString(info.declaration.parameters[0].name) : undefined;
     }
 
-    export function getTextOfPropertyName(name: PropertyName): string {
+    export function getTextOfPropertyName(name: PropertyName): UnderscoreEscapedString {
         switch (name.kind) {
             case SyntaxKind.Identifier:
                 return (<Identifier>name).text;
             case SyntaxKind.StringLiteral:
             case SyntaxKind.NumericLiteral:
-                return (<LiteralExpression>name).text;
+                return (<LiteralExpression>name).text as UnderscoreEscapedString;

```

Comment:
I think you're right, this is probably unsafe, given where this is used. I'll add the escapes, rather than casts.